runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -300,18 +300,16 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 	 */
 	MMINLINE UDATA
 	getDataSizeInBytes(J9Class *clazzPtr, UDATA numberOfElements)
-	{
-		UDATA alignedSize = UDATA_MAX;
-		J9ROMArrayClass *romArrayClass = (J9ROMArrayClass *)clazzPtr->romClass;
-		UDATA shift = romArrayClass->arrayShape & 0x0000FFFF;
-		UDATA size = numberOfElements << shift;
-		if ((size >> shift) == numberOfElements) {
```

Comment:
This check has been removed from the new code.

Reply:
Yea, I was wondering when will this check ever return false...

Reply:
Ask @dmitripivkine - it may have something to do with overflow detection.

Reply:
This is overflow detection on 32-bit platforms

Reply:
Will it still need to be kept? If so, would this be a good substitution? 

```
	MMINLINE UDATA
	getDataSizeInBytes(J9Class *clazzPtr, UDATA numberOfElements)
	{	
		UDATA stride = J9ARRAYCLASS_GET_STRIDE(clazzPtr);
		UDATA size = numberOfElements * stride;
		UDATA alignedSize = UDATA_MAX;
		if ((size/stride) == numberOfElements) {
			alignedSize = MM_Math::roundToSizeofUDATA(size);
			if (alignedSize < size) {
				alignedSize = UDATA_MAX;
			}
		}
		return alignedSize;

	}
```

Reply:
Logically this is code should work. However my concern here is an introduction of real division operation might be expensive some platforms. There is no way for compiler to replace it to shift any kind.
@charliegracie Any thoughts?

Reply:
To check for overflow, simply do:

orig = val
val = val * size
if (val < orig) overflow

Reply:
It would not work for all cases. For example:
orig = 0x7FFFFFFF
size = 0x18
val = (orig * size) = 0x(B)FFFFFFE8

val is larger then orig but this is overflow

Reply:
I see a few solutions but all of them not better then using division

Reply:
I've decided to stick with using division